### PR TITLE
added partition aware event handlers

### PR DESCRIPTION
**Pull request by perokvist**

### CODE REVIEW

**Comment by vullnetyy** [comment]
@perokvist I've added a commit to this to make the change to Func<string, Func<IReadOnlyCollection<EventData>, CancellationToken, Task>> for handleEvents in the stateless service too.

I approve this PR now, but am waiting for your input before I merge.

**Review by vullnetyy — APPROVED** [review]

**Comment by perokvist** [comment]
@vullnetyy  great, partitionId was supported, but not used in the Receiver Service.

File: CMP.ServiceFabricReceiver.Stateful/EventProcessor.cs
Comment (allrameest) [inline CMP.ServiceFabricReceiver.Stateful/EventProcessor.cs]: It's hard to know what these string parameters is. Would be good to use either a named delegate or another more specific type.

Comment (perokvist) [reply CMP.ServiceFabricReceiver.Stateful/EventProcessor.cs]: yeah, might be an improvement. Is kind of internal.
I also think the aspects of processing should be done by composition, meaning removing the "retry once" logic, to be able to build this trough composing aspects when registering the eventhandler.

Comment (vullnetyy) [reply CMP.ServiceFabricReceiver.Stateful/EventProcessor.cs]: done

**Pull request merged**
